server/src/main/java/io/crate/planner/optimizer/rule/MoveFilterBeneathProjectSet.java

```diff
@@ -62,12 +63,17 @@ public LogicalPlan apply(Filter filter,
                              Rule.Context ruleContext) {
         var projectSet = captures.get(projectSetCapture);
 
+        HashSet<Symbol> outputs = new HashSet<>();
+        for (Symbol output : projectSet.standaloneOutputs()) {
+            outputs.addAll(extractColumns(output));
```

Comment:
Extract underlying columns from outputs from project-set and compare them against the underlying columns from the filter.

Reply:
Do I get it right, that each output in `standaloneOutputs` can be a function or anything else that in turn has many output columns and we want to include them all?

Bit confused that `projectSet` has `standaloneOutputs() `and `outputs()`. 

Judging by the naming, it sounds like each standalone-output must be already "standalone" which I understand as "nothing to extract"

But looking at ctor, only table functions are filtered, so can indeed have something to extract.
```
  List<Symbol> standalone = result.outputs().stream()
                .filter(x -> !(x instanceof Function fn && fn.signature().getType() == FunctionType.TABLE))
                .toList();
```

Just curious: what Symbol in `standaloneOuputs` actually produces some extra outputs via `extractColumns`?

Reply:
In this case the filter-query contains an `AliasedSymbol` which underneath contains a `Reference`. To be able to match on the outputs of the `ProjectSet` which are `References`, `ScopedSymbols` etc. the underlying `References` from these also need to extracted.